modules/EPMC.py

```diff
@@ -0,0 +1,183 @@
+    ##
+    global spark
+    local = True
+
+    if args.local:
+        sparkConf = (SparkConf()
+                 .set("spark.driver.memory", "15g")
+                 .set("spark.executor.memory", "15g")
+                 .set("spark.driver.maxResultSize", "0")
+                 .set("spark.debug.maxToStringFields", "2000")
+                 .set("spark.sql.execution.arrow.maxRecordsPerBatch", "500000")
+                 )
+        spark = (
+            SparkSession.builder
+                .config(conf=sparkConf)
+                .master('local[*]')
+                .getOrCreate()
+        )
+    else:
+        sparkConf = (SparkConf()
+                 .set("spark.driver.maxResultSize", "0")
+                 .set("spark.debug.maxToStringFields", "2000")
+                 .set("spark.sql.execution.arrow.maxRecordsPerBatch", "500000")
+                 )
+        spark = (
+            SparkSession.builder
+                .config(conf=sparkConf)
+                .getOrCreate()
+        )
+    logging.info(f'Spark version: {spark.version}')
+
+    ##
+    ## Log parameters:
+    ##
+    logging.info(f'Cooccurrence file: {cooccurrenceFile}')
+    logging.info(f'Output file: {out_file}')
+    logging.info(f'Generating evidence:')
+
+    filtered_target_terms = ["TEC", "TECS", "Tec", "tec", "'", "(", ")", "-", "-S", "S", "S-", "SS", "SSS",
+         "Ss", "Ss-", "s", "s-", "ss", "U3", "U6", "u6", "SNORA70", "U2", "U8"]
+
+    ##
+    ## Load/filter datasets
+    ##
+    # partitionKeys = ['pmid', 'targetFromSourceId', 'diseaseFromSourceMappedId']
+    # w = Window.partitionBy(*partitionKeys)
+    filtered_cooccurrence_df = (
+        # Reading file:
+        spark.read.parquet(cooccurrenceFile)
+
+        # Filtering for diases/target cooccurrences:
+        .filter(
+            (col('type') == "GP-DS") & 
+            (col('isMapped') == True) & 
+            (col('pmid').isNotNull()) & 
+            (length(col('text')) < 600) &
+            (col('label1').isin(filtered_target_terms) == False)
+        )
+        # Renaming columns:
+        .withColumnRenamed("keywordId1", "targetFromSourceId")
+        .withColumnRenamed("keywordId2", "diseaseFromSourceMappedId")
+        .withColumnRenamed("label1", "targetFromSource")
+        .withColumnRenamed("label2", "diseaseFromSource")
+
+        .withColumn('tmp',col('pmid').cast(StringType()))
+    )
+
+    # Report on the number of diseases, targets and associations:
+    logging.info(f'Number of publications: {filtered_cooccurrence_df.select(col("tmp")).distinct().count()}')
+    logging.info(f'Number of targets: {filtered_cooccurrence_df.select(col("targetFromSourceId")).distinct().count()}')
+    logging.info(f'Number of diseases: {filtered_cooccurrence_df.select(col("diseaseFromSourceMappedId")).distinct().count()}')
+    logging.info(f'Number of associations: {filtered_cooccurrence_df.select(col("diseaseFromSourceMappedId"), col("targetFromSourceId")).dropDuplicates().count()}')
```

Comment:
Also some literals:

```suggestion
    logging.info(f'Number of publications: {filtered_cooccurrence_df.select(col('tmp')).distinct().count()}')
    logging.info(f'Number of targets: {filtered_cooccurrence_df.select(col('targetFromSourceId')).distinct().count()}')
    logging.info(f'Number of diseases: {filtered_cooccurrence_df.select(col('diseaseFromSourceMappedId')).distinct().count()}')
    logging.info(f'Number of associations: {filtered_cooccurrence_df.select(col('diseaseFromSourceMappedId'), col('targetFromSourceId')).dropDuplicates().count()}')
```

Reply:
It won't work in f-stings.  It needs to be resolved by this:
```python
logging.info(f'Number of publications: {filtered_cooccurrence_df.select(col("tmp")).distinct().count()}')
```
to this:

```python
logging.info(f"Number of publications: {filtered_cooccurrence_df.select(col('tmp')).distinct().count()}")
```

Reply:
> It won't work in f-stings.

Indeed. In this case this wasn't a particularly well thought through suggestion from my side. Apologies and thanks for noticing :)